### ISSUE TITLE
tests failing with node 4.2.1

### ISSUE THREAD

**Issue by terceiro** [mbostock/smash#25]
originally found [on Debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=802111). I can also reproduce this in clean system with nothing but nodejs and npm installed.

```
$ git remote -v
origin	git@github.com:mbostock/smash.git (fetch)
origin	git@github.com:mbostock/smash.git (push)
$ git clean -dxf
$ node --version
v4.2.1
$ npm --version
1.4.21
$ npm install
queue-async@1.0.7 node_modules/queue-async

optimist@0.3.7 node_modules/optimist
└── wordwrap@0.0.3

vows@0.7.0 node_modules/vows
├── diff@1.0.8
└── eyes@0.1.8
$ npm run test

> smash@0.0.14 test /tmp/smash
> vows; echo

········· ✗✗✗ ·····✗✗······· ·····✗✗······· ··········· ········✗···········  
   
   
    load on a simple file 
      ✗ returns the evaluated expression 
        » An unexpected error was caught: TypeError: sandbox must be an object 
    at TypeError (native) 
    at Object.exports.createContext (vm.js:30:22) 
    at ContextifyScript.Script.runInNewContext (vm.js:17:25) 
    at Object.exports.runInNewContext (vm.js:49:17) 
    at PassThrough.<anonymous> (/tmp/smash/lib/smash/load.js:14:27) 
    at emitNone (events.js:72:20) 
    at PassThrough.emit (events.js:166:7) 
    at PassThrough.options.Emitter.emit (/tmp/smash/node_modules/vows/lib/vows.js:237:24) 
    at endReadableNT (_stream_readable.js:903:12) 
    at doNTCallback2 (node.js:439:9) 
    at process._tickCallback (node.js:353:17) 
   
   
      ✗ does not pollute the global namespace 
        » An unexpected error was caught: TypeError: sandbox must be an object 
    at TypeError (native) 
    at Object.exports.createContext (vm.js:30:22) 
    at ContextifyScript.Script.runInNewContext (vm.js:17:25) 
    at Object.exports.runInNewContext (vm.js:49:17) 
    at PassThrough.<anonymous> (/tmp/smash/lib/smash/load.js:14:27) 
    at emitNone (events.js:72:20) 
    at PassThrough.emit (events.js:166:7) 
    at PassThrough.options.Emitter.emit (/tmp/smash/node_modules/vows/lib/vows.js:237:24) 
    at endReadableNT (_stream_readable.js:903:12) 
    at doNTCallback2 (node.js:439:9) 
    at process._tickCallback (node.js:353:17) 
   
   
    load with an object literal expression 
      ✗ returns the evaluated expression 
        » An unexpected error was caught: TypeError: sandbox must be an object 
    at TypeError (native) 
    at Object.exports.createContext (vm.js:30:22) 
    at ContextifyScript.Script.runInNewContext (vm.js:17:25) 
    at Object.exports.runInNewContext (vm.js:49:17) 
    at PassThrough.<anonymous> (/tmp/smash/lib/smash/load.js:14:27) 
    at emitNone (events.js:72:20) 
    at PassThrough.emit (events.js:166:7) 
    at PassThrough.options.Emitter.emit (/tmp/smash/node_modules/vows/lib/vows.js:237:24) 
    at endReadableNT (_stream_readable.js:903:12) 
    at doNTCallback2 (node.js:439:9) 
    at process._tickCallback (node.js:353:17) 
   
   
    readAllImports on a file with that imports a file that does not exist 
      ✗ throws an error with the expected message 
        »        
        actual expected 
         
        ENOENT: no such file or directory, open 'test/data/not-found.js' 
         // readAllImports-test.js:44 
   
    readAllImports on a file with that imports a file that does not exist with --ignore-missing 
      ✗ returns the expected imports 
        » An unexpected error was caught: Error: ENOENT: no such file or directory, open 'test/data/not-found.js' 
    at Error (native) 
   
   
    readGraph on a file with that imports a file that does not exist 
      ✗ throws an error with the expected message 
        »        
        actual expected 
         
        ENOENT: no such file or directory, open 'test/data/not-found.js' 
         // readGraph-test.js:50 
   
    readGraph on a file with that imports a file that does not exist with --ignore-missing 
      ✗ returns the empty array 
        » An unexpected error was caught: Error: ENOENT: no such file or directory, open 'test/data/not-found.js' 
    at Error (native) 
   
   
    smash on a file with that imports a file that does not exist 
      ✗ produces the expected error message 
        »        
        actual expected 
         
        ENOENT: no such file or directory, open 'test/data/not-found.js' 
         // smash-test.js:56 
  ✗ Errored » 63 honored ∙ 3 broken ∙ 5 errored (0.156s) 
```

**Comment by mbostock** [comment]
I just edited the repository description to indicate that SMASH is deprecated in favor of [Rollup](https://github.com/rollup/rollup). It’s unlikely I will address this issue. Sorry.

**Comment by thomasgoirand** [comment]
Hi. Since this package affects many dependencies in Debian, including nodejs itself, and everything which uses uglify as a consequence, it'd be very nice if a fix was done, at least until we can (have the time to) switch to Rollup later on.

**Comment by kapouer** [comment]
@thomasgoirand @terceiro please check my pull requests and include them in node-smash debian package.

**Comment by mbostock** [comment]
I’ve accepted your patches and published 0.0.15. Thank you!

**Comment by terceiro** [comment]
thanks @kapouer for working on it and @mbostock for making a release!
